### ISSUE TITLE
For DOI search translation, use web translation on resolved DOI page before using other search translators

### ISSUE THREAD

**Issue by dstillman** [zotero/zotero#1562]
@adam3smith, @zuphilip, thoughts on resolving the DOI and performing web translation as a first step before using the individual search translators? @mrtcode has been working with a lot of metadata, and he's noticed that, at least with Crossref DOIs, the metadata we get from publisher pages is often higher quality than the data from the Crossref API (e.g., Abstract, obviously, but also creators and maybe some other things).

If web translation failed, it would fall back to using the DOI search translators.

There are some technical issues here that will need to be resolved before we can do this, but any objections in principle?

**Comment by adam3smith** [comment]
I'm a bit surprised by the non-abstract issues (I guess a testament to how good the translators are), but no objections

**Comment by zuphilip** [comment]
This is certainly an interesting idea and might help to receive better metadata in some cases. However, I am not sure that this would give us better data in _all_ cases, e.g. when embedded metadata translator just find some rudimentary data on the publisher's website. I don't have much experience with metadata form DOI agencies besides Crossref or DataCite.

Publisher's have one or several DOI prefixes. Thus, it might be possible to decide to check the website with its translator first only for a subset of all publishers resp. always exept for some subset of publishers.

Another big advantage for checking the publisher's website first, is that there are chances that the fulltext is attached as a PDF. This will never happen when calling the DOI translator.

Would calling the publisher's website in the background and then perform a translator be comparable to testing in Scaffold? This testing in Scaffold does not always work because of some dynamic website loading techniques etc. E.g. I cannot run tests for ScienceDirect translator in Scaffold. Moreover, there could be differences in the time it takes to open a website compared to call an API.

**Comment by dstillman** [comment]
> However, I am not sure that this would give us better data in all cases, e.g. when embedded metadata translator just find some rudimentary data on the publisher's website.

Yeah, that's a good point. I think @mrtcode wants to have some logic here to determine when to try the APIs even if the web translation nominally succeeds.

> Publisher's have one or several DOI prefixes. Thus, it might be possible to decide to check the website with its translator first only for a subset of all publishers resp. always exept for some subset of publishers.

Interesting idea, though since translators might vary in quality over time, we probably want a general mechanism for this.

> Would calling the publisher's website in the background and then perform a translator be comparable to testing in Scaffold? This testing in Scaffold does not always work because of some dynamic website loading techniques

Answered in the referenced issue above.
